### PR TITLE
Adding helpers to make gRPC stubs from a client.

### PR DESCRIPTION
**Pull request by dhermes**
~~NOTE: Uses #1152 as diffbase (though doesn't need to, as I discovered while writing)~~

### CODE REVIEW

**Comment by dhermes** [comment]
@tseaver I just rebased this. PTAL.

File: gcloud/bigtable/client.py [hunk 1]
```diff
@@ -86,12 +92,22 @@ class Client(_ClientFactoryMixin, _ClientProjectMixin):
...
 
+    :type user_agent: str
+    :param user_agent: (Optional) The user agent to be used with API request.
+                       Defaults to :const:`DEFAULT_USER_AGENT`.
+
+    :type timeout_seconds: int
+    :param timeout_seconds: Number of seconds for request time-out. If not
+                            passed, defaults to
+                            :const:`DEFAULT_TIMEOUT_SECONDS`.
+
```
Comment (tseaver) [inline gcloud/bigtable/client.py]: Where are these values actually used?  Is that in code yet-to-be-added?

Comment (dhermes) [inline gcloud/bigtable/client.py]: Yes it is yet-to-be-added.

- `user_agent` gets sent in request metadata
- `timeout_seconds` is a rule to give up on a request if it doesn't succeed (like a socket timeout)

Comment (tseaver) [inline gcloud/bigtable/client.py]: I see `user_agent` being used above now.  Is the code which actually makes the request (and therefore which uses the timeout) generated?

Comment (dhermes) [inline gcloud/bigtable/client.py]: Most of it happens in gRPC, some of it happens in generated code as well.

We make a gRPC stub object:
https://github.com/dhermes/gcloud-python-bigtable/blob/21aa4a73df0fa7529285271e9fe2cc28b1e992e7/gcloud_bigtable/_helpers.py#L296-L299

The `stub_factory` is specific to the service. For example
https://github.com/dhermes/gcloud-python-bigtable/blob/21aa4a73df0fa7529285271e9fe2cc28b1e992e7/gcloud_bigtable/client.py#L55-L56
is a possible choice (`bigtable_table_service_pb2.early_adopter_create_BigtableTableService_stub`).

Comment (dhermes) [inline gcloud/bigtable/client.py]: Then the stubs are called with the specific RPC methods:
https://github.com/dhermes/gcloud-python-bigtable/blob/21aa4a73df0fa7529285271e9fe2cc28b1e992e7/gcloud_bigtable/client.py#L607-L608

But we set the stubs to a permanently entered state (they are context managers):
https://github.com/dhermes/gcloud-python-bigtable/blob/21aa4a73df0fa7529285271e9fe2cc28b1e992e7/gcloud_bigtable/client.py#L503-L521

**Comment by tseaver** [comment]
LGTM

**Pull request merged**